Bio/PDB/Structure.py

```diff
@@ -63,3 +94,85 @@ def internal_to_atom_coordinates(self, verbose: bool = False) -> None:
         """
         for chn in self.get_chains():
             chn.internal_to_atom_coordinates(verbose)
+
+    def _find_residue_by_id(self, chain: Chain, id_: int):
+        """
+        Return the residue with `id_` from the `chain`.
+
+        This is a workaround required due to Biopython's residue indexing e.g.
+        ('H_GLC', 100, 'A') instead of 100. For more info please refer to the
+        section "What is a residue id?" at
+        https://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ
+        """
+        for residue in chain.get_residues():
```

Comment:
This is going to be quite expensive.. but I assume we don't have a better solution unless we somehow cache things.